src/vs/workbench/contrib/terminal/browser/widgets/terminalQuickFixWidget.ts

```diff
@@ -0,0 +1,265 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ActionSet, ActionShowOptions, ListMenuItem, stripNewlines } from 'vs/base/browser/ui/baseActionWidget/baseActionWidget';
+import { IAction } from 'vs/base/common/actions';
+import { Disposable, IDisposable } from 'vs/base/common/lifecycle';
+import { ActionItemRenderer, ActionList, ActionListItemKind, ActionWidget, HeaderRenderer, IRenderDelegate } from 'vs/editor/contrib/actionWidget/browser/actionWidget';
+import { localize } from 'vs/nls';
+import { IContextKeyService, RawContextKey } from 'vs/platform/contextkey/common/contextkey';
+import { IContextViewService } from 'vs/platform/contextview/browser/contextView';
+import { IInstantiationService } from 'vs/platform/instantiation/common/instantiation';
+import { CodeActionKind } from 'vs/editor/contrib/codeAction/common/types';
+import { Codicon } from 'vs/base/common/codicons';
+import { IKeybindingService } from 'vs/platform/keybinding/common/keybinding';
+import { Action2, registerAction2 } from 'vs/platform/actions/common/actions';
+import { KeybindingWeight } from 'vs/platform/keybinding/common/keybindingsRegistry';
+import { KeyCode, KeyMod } from 'vs/base/common/keyCodes';
+import { ICommandService } from 'vs/platform/commands/common/commands';
+import { ITelemetryService } from 'vs/platform/telemetry/common/telemetry';
+
+const acceptSelectedTerminalQuickFixCommand = 'acceptSelectedTerminalQuickFixCommand';
+export const previewSelectedTerminalQuickFixCommand = 'previewSelectedTerminalQuickFixCommand';
+const weight = KeybindingWeight.EditorContrib + 1000;
+
+export class TerminalQuickFix extends Disposable {
+	action?: IAction;
+	disabled?: boolean;
+	title?: string;
+	constructor(action?: IAction, title?: string, disabled?: boolean) {
+		super();
+		this.action = action;
+		this.disabled = disabled;
+		this.title = title;
+	}
+}
+
+export const Context = {
+	Visible: new RawContextKey<boolean>('terminalQuickFixMenuVisible', false, localize('terminalQuickFixMenuVisible', "Whether the terminal quick fix menu is visible"))
+};
+
+export class TerminalQuickFixWidget extends ActionWidget<TerminalQuickFix> {
+
+	private static _instance?: TerminalQuickFixWidget;
+
+	public static get INSTANCE(): TerminalQuickFixWidget | undefined { return this._instance; }
+
+	public static getOrCreateInstance(instantiationService: IInstantiationService): TerminalQuickFixWidget {
+		if (!this._instance) {
+			this._instance = instantiationService.createInstance(TerminalQuickFixWidget);
+		}
+		return this._instance;
+	}
+
+	constructor(
```

Comment:
There's nothing stopping `TerminalQuickFixWidget.ctor` from being invoked, is there a large amount of overhead in creating it or something? Seems like it would be better to recreate it when it's invoked to me?

Reply:
I was just following suit w the `CodeActionWidget`